stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -70,15 +70,18 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
 
   @ChromeDevtoolsMethod
   public JsonRpcResult getDocument(JsonRpcPeer peer, JSONObject params) {
-    Object rootElement = mDOMProvider.getRootElement();
-    if (rootElement == null) {
-      return null;
-    }
-
-    Node rootNode = createNodeForElement(rootElement);
+    final GetDocumentResponse result = new GetDocumentResponse();
+
+    mDOMProvider.postAndWait(new Runnable() {
+       @Override
+       public void run() {
+         Object rootElement = mDOMProvider.getRootElement();
+         if (rootElement != null) {
+           result.root = createNodeForElement(rootElement);
```

Comment:
That doesn't seem necessary though? It's safe because postAndWait() doesn't return until the Runnable completes. Using an AtomicReference wouldn't gain us real safety, and then this code might look like it's being unnecessarily over-cautious due to the developer not being sure about the safety of not using AtomicReference in this case.